src/main/java/logic/CompareFile.java

```diff
@@ -0,0 +1,153 @@
+package logic;
+
+import data.FileDifference;
+import data.ResultData;
+import data.StatusFile;
+import logic.ListFileCreator.ListOfFilesFromPathCreator;
+import logic.unzip.UnzipFileToDirectoryController;
+import logic.unzip.UnzipFileToDirectoryable;
+import ui.Messageble;
+import ui.UserInterface;
+import java.io.*;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+public class CompareFile implements FileStrategy {
+    private final String PDF_FILE = "pdf";
+    private final String DOCX_FILE = "docx";
+    private final String XLSX_FILE = "xlsx";
+    private static final int GREATER_THAN_ZERO = 0;
+    private Messageble messageble;
+    private UserInterfaceController userInterfaceController;
+    private KeyboardReader keyboardReader;
+    private UnzipFileToDirectoryable unzipFileToDirectoryable;
+    private File sourceFile;
+    private File fileToCompare;
+
+    public CompareFile(){
+        this.messageble = new UserInterface();
+        this.keyboardReader = new KeyboardReader(new BufferedReader(new InputStreamReader(System.in)));
+        this.userInterfaceController = new UserInterfaceController(new GetFileFromConsole());
+        unzipFileToDirectoryable = new UnzipFileToDirectoryController();
+        getFilesFromUser();
+    }
+
+    public CompareFile(File sourceFile, File fileToCompare) {
+        this.messageble = new UserInterface();
+        this.keyboardReader = new KeyboardReader(new BufferedReader(new InputStreamReader(System.in)));
+        this.userInterfaceController = new UserInterfaceController(new GetFileFromConsole());
+        unzipFileToDirectoryable = new UnzipFileToDirectoryController();
+        this.sourceFile = sourceFile;
+        this.fileToCompare = fileToCompare;
+    }
+
+    @Override
+    public ResultData perform() {
+        try {
+            Path tempDirSource = Files.createTempDirectory("");
+            Path tempDirToCompare = Files.createTempDirectory("");
+            unzipFileToDirectoryable.unzip(sourceFile.toPath(), tempDirSource);
+            unzipFileToDirectoryable.unzip(fileToCompare.toPath(), tempDirToCompare);
+            List<File> sourceFiles = new ListOfFilesFromPathCreator().getListOfFile(tempDirSource.toString());
+            List<File> filesToCompare = new ListOfFilesFromPathCreator().getListOfFile(tempDirToCompare.toString());
+
+            List<FileDifference> fileDifferences = getListOfDifferences(sourceFiles, filesToCompare);
+
+        } catch (IOException ioException) {
+            System.out.println(ioException.fillInStackTrace());
+        }
+
```

Comment:
hej albo przenieś to do bloku try albo do finally

Reply:
Zrobione